samples/tutorial/Tutorial1.md

```diff
@@ -0,0 +1,245 @@
+
+We can finally run the app again! It will look exactly the same as before, but now it is powered by our workflow.
+
+## Driving the UI from Workflow State
+
+Right now, the workflow isn't handling any of the events from the UI. Let's update it to be responsible for the login name as well as the action when the login button is pressed.
+
+### State
+
+All workflows have a `State` type that represents the internal state of the workflow. This should be all of the data for which *this* workflow is _responsible_. It usually corresponds to the state for the UI.
+
+Let's model the first part of state that we want to track: the login `name`. Update the `State` type to include a name property. We will also need to update `initialState` to give an initial value:
+
+```kotlin
+object WelcomeWorkflow : StatefulWorkflow<Unit, State, Output, WelcomeScreen>() {
+
+  data class State(
+    val name: String
+  )
+
+  // …
+
+  override fun initialState(
+    props: Unit,
+    snapshot: Snapshot?
+  ): State = State(name = "")
+
+  // …
+}
+
+```
+
+Now that we have the state modeled, we'll send it to the UI every time a render pass happens. The text field will overwrite its value with whatever was provided.
+
+If you run the app again, you'll see that it still behaves the same, letting your type into the name field. This is because we have only rendered the screen once.
+
+You may have noticed that your workflow only has access to its `State` in a few functions, and even then in many cases it is read-only. This is intentional. The Workflow infrastructure manages the state for the various workflows that are running and triggers a re-render when appropriate. In order to update the workflow's internal state, we need to add an "Action",
+
+### Actions
+
+Actions define how a workflow handles events received from the outside world, such as UI events (e.g. button presses), network requests, data stores, etc. Generally a `WorkflowAction` is a function which defines a particular state transition or side effect. Typically you define a `WorkflowAction` by writing a function that returns the actual action. This makes it easy to pass parameters to your actions. The action itself may not execute right away.
+
+Add a function called `onNameChanged` to update our internal state:
+
+```kotlin
+  private fun onNameChanged(name: String) = action {
+    state = state.copy(name = name)
+  }
+```
+
+The `action` function is a shorthand for implementing the `WorkflowAction` class yourself. You could also write:
+
+```kotlin
+  private fun onNameChanged(name: String) =
+    object : WorkflowAction<Unit, State, Nothing>() {
+      override fun Updater.apply() {
+        this.state = this.state.copy(name = name)
+      }
+    }
+```
+
+We need to send this action back to the workflow any time the name changes. Update the `render` method to send it through a sink back to the workflow whenever the `onNameChanged` closure is called:
```

Comment:
While 'sync' is a general concept, it could use a quick sidebar/explanation here.

Reply:
Added a bit.